zorg/buildbot/builders/FlangBuilder.py

```diff
@@ -71,4 +92,58 @@ def getFlangOutOfTreeBuildFactory(
        stage_name="flang",
        **kwargs)
 
+    ## Build Flang-RT as a standalone runtime
+    flang_rt_obj_dir = "build_flang-rt"
+
+    flang_rt_cmake_args = ["-GNinja"]
+    if flang_rt_extra_configure_args:
+        flang_rt_cmake_args += flang_rt_extra_configure_args
+
+    # Use LLVM from the getCmakeWithNinjaBuildFactory step.
+    flang_rt_cmake_args += [
+        util.Interpolate(f"-DLLVM_BINARY_DIR=%(prop:builddir)s/{f.obj_dir}"),
+        "-DLLVM_ENABLE_RUNTIMES=flang-rt",
+    ]
+
+    # Use the Fortran compiler from the previous step.
+    flang_rt_cmake_args += [
+        util.Interpolate(
+            f"-DCMAKE_Fortran_COMPILER=%(prop:builddir)s/{flang_obj_dir}/bin/flang"
+        ),
+        "-DCMAKE_Fortran_COMPILER_WORKS=ON",
+    ]
+
+    f.addStep(
+        steps.RemoveDirectory(
+            name=f"clean-{flang_rt_obj_dir}-dir",
+            dir=flang_rt_obj_dir,
+            haltOnFailure=False,
+            flunkOnFailure=False,
+            doStepIf=cleanBuildRequested,
+        )
+    )
+
+    f.addStep(
+        CmakeCommand(
+            name="cmake-configure-flang-rt",
+            haltOnFailure=True,
+            description=["CMake", "configure", "Flang-RT"],
+            options=flang_rt_cmake_args,
+            path=LLVMBuildFactory.pathRelativeTo(
+                os.path.join(f.monorepo_dir, "runtimes"), flang_rt_obj_dir
+            ),
+            env=env,
+            workdir=flang_rt_obj_dir,
+            **kwargs,
+        )
+    )
+
+    addNinjaSteps(
+       f,
+       obj_dir=flang_rt_obj_dir,
+       checks=['check-flang-rt'],
```

Comment:
I would prefer to have build and test as separate steps. A single step is ok but it's always a bit of extra thinking when a build problem makes a check step red.

Reply:
`addNinjaSteps` is adding multiple steps, indicated by its plural name. Or see its code: https://github.com/llvm/llvm-zorg/blob/main/zorg/buildbot/builders/UnifiedTreeBuilder.py#L215-L242.

Reply:
I see, so this will `ninja` then `ninja check-flang-rt`. That's fine.